app/updatemanager.go

```diff
@@ -401,6 +404,32 @@ func (c *ControlMapPool) QueryAndUpdate(state string) (action string) {
 	return "continue"
 }
 
+func (c *ControlMapPool) NextControlMapHalfTime(ID string) time.Time {
+	m := []*updatecontrolmap.UpdateControlMap{}
+	query(c.Pool,
+		// Collector
+		func(u *updatecontrolmap.UpdateControlMap) {
+			m = append(m, u)
+		},
+		// Predicates
+		func(u *updatecontrolmap.UpdateControlMap) bool {
+			return u.ID == ID
+		},
+	)
+	// Return the time of the map with the closest time to halfway expired
+	if len(m) == 0 {
+		return time.Now().Add(time.Duration(c.updateControlMapExpirationTimeSeconds) / 2)
```

Comment:
Why do you need a time at all in this case? Can't you just return some kind of "empty" error instead? If I understand it correctly, this could save several things:

* You don't need to introduce the `updateControlMapExpirationTimeSeconds` variable in the `ControlMap` struct.
* If you check for the error in the caller, you can avoid a call to the backend if we don't have any update maps.

Reply:
Good idea!